ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -340,18 +340,12 @@ impl LoadArgInstruction {
 pub struct LoadGlobalInstruction {
     /// The pointer to load from.
     global_idx: GlobalIdx,
-    /// The type of the pointee.
-    ty_idx: TypeIdx,
```

Comment:
How comes this goes?

Reply:
When we emit the instruction manually from a global (which currently doesn't have a type), what should we put here? I believe the return type of all globals is `ptr` anyway, so there's probably no point. We do need to somehow work out the actual type later on when codegenning this, but I believe that can be retrieved from the use site. E.g. when we pass the ptr into a call, the call arg will tell us what type it expects.

Reply:
> I believe the return type of all globals is ptr anyway

That depends on the semantics of your load.

In LLVM you'd always get the type that's pointed to.

Let's roll with what you have and see what happens later.

Reply:
(I wonder if "load global" is a load at all?)

Reply:
I assumed loading a global returns the ptr stored in the global, not the data that ptr points to. But I might be wrong. It's hard to say without looking at examples I guess.

Agreed, let's deal with it when we need to codegen to know exactly what we are dealing with,

Reply:
> (I wonder if "load global" is a load at all?)

Yes, I think in our case loading a global is more akin to a lookup where a global name maps to some pointer.

Reply:
In the meantime, here's an example of how LLVM does it:
https://llvm.org/docs/LangRef.html#id210

i.e. a load is a bit like a deref.

Reply:
No preference. Though maybe waiting until we know for sure how globals need to be codegenned isn't a bad idea.